src/components/MarkdownLayout/TableOfContents/getGithubUrl.tsx

```diff
@@ -4,18 +4,35 @@ import { graphql, useStaticQuery } from 'gatsby';
 import { SolutionInfo } from '../../../models/solution';
 import { ModuleInfo } from '../../../models/module';
 
-export default function getGithubUrl() {
+export default function getInfoToEdit() {
   const moduleLayoutInfo = useContext(MarkdownLayoutContext);
   const markdownInfo = moduleLayoutInfo.markdownLayoutInfo;
   const relativePath = markdownInfo.fileRelativePath;
-  let githubUrl = 'https://github.com/cpinitiative/usaco-guide/blob/master/';
-
+  const githubUrlPrefix =
+    'https://github.com/cpinitiative/usaco-guide/blob/master/';
+  const rawGithubUrlPrefix =
+    'https://raw.githubusercontent.com/cpinitiative/usaco-guide/master/';
+  let suffix = '';
   if (markdownInfo instanceof ModuleInfo) {
     // /\d_.*/.test(relativePath)
-    githubUrl += 'content/' + relativePath;
+    suffix = 'content/' + relativePath;
   } else {
-    githubUrl += 'solutions/' + relativePath;
+    suffix = 'solutions/' + relativePath;
   }
-  githubUrl = githubUrl.replace(' ', '%20'); // is there some function that can automatically escape chars?
+
+  const escapeSpaces = url => url.replace(' ', '%20');
+  const githubUrl = escapeSpaces(githubUrlPrefix + suffix);
+  const rawGithubUrl = escapeSpaces(rawGithubUrlPrefix + suffix);
+  // is there some function that can automatically escape chars?
+
+  // https://www.scrapingbee.com/blog/web-scraping-javascript/
+  // https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
+  const getContents = async () => {
+    const data = await fetch(rawGithubUrl);
+    const text = await data.text();
+    console.log('TEXT', text); // ok what do I do with this
+  };
+
+  getContents();
```

Comment:
maybe move this to LiveUpdate? And in the `LinksToEdit` component, rather than use an OutboundLink, use a regular internal Gatsby Link pointing to `/liveupdate`, passing in a state of the Github URL: https://www.gatsbyjs.com/docs/location-data-from-props/#example-of-providing-state-to-a-link-component

And then in the LiveUpdate component, based on the incoming state, fetch the data from Github and populate the content of the editor?